discovery-provider/tests/test_ipfs_functionality.py

```diff
@@ -40,3 +40,50 @@ def test_ipfs(app):
     for key in test_metadata_object:
         assert key in ipfs_metadata
         assert test_metadata_object[key] == ipfs_metadata[key]
+
+# test for ipfs_lib.py, multihash_is_directory
+class TestMultihashIsDirectory:
+    # Invoke multihash_is_directory with a invalid cid, verify that it throws an 'invalid multihash' error
+    def test_invalid_cid(self, app):
+        ipfs_peer_host = app.config["ipfs"]["host"]
+        ipfs_peer_port = app.config["ipfs"]["port"]
+
+        # Instantiate IPFS client from src lib
+        ipfsclient = IPFSClient(ipfs_peer_host, ipfs_peer_port, [])
+
+        try:
+            ipfsclient.multihash_is_directory('Qmfake')
+        except Exception as e:
+            assert 'invalid multihash' in str(e)
+
+    # Invoke multihash_is_directory with cat function handler that returns some other response
+    def test_valid_cid_directory(self, app):
+        ipfs_peer_host = app.config["ipfs"]["host"]
+        ipfs_peer_port = app.config["ipfs"]["port"]
+
+        # Instantiate IPFS client from src lib
+        ipfsclient = IPFSClient(ipfs_peer_host, ipfs_peer_port, [])
+
+        def cat_handler_dir(cid, s, e, timeout):
+            raise Exception('this dag node is a directory')
+        ipfsclient._api.cat = cat_handler_dir
+
+        # Invoke audius-ipfs with a invalid cid, verify that it throws an 'invalid multihash' error
+        is_directory = ipfsclient.multihash_is_directory('QmVmEZnQr49gDtd7xpcsNdmgrtRTT5Te2x27KbTRPNapqy')
+        assert is_directory is True
+
+    # Invoke multihash_is_directory with cat function handler that returns `this dag node is a directory error`
+    def test_valid_cid_not_directory(self, app):
+        ipfs_peer_host = app.config["ipfs"]["host"]
+        ipfs_peer_port = app.config["ipfs"]["port"]
+
+        # Instantiate IPFS client from src lib
+        ipfsclient = IPFSClient(ipfs_peer_host, ipfs_peer_port, [])
+
+        def cat_handler_not_dir(cid, s, e, timeout):
+            return 'not a directory'
+        ipfsclient._api.cat = cat_handler_not_dir
+
+        # Invoke audius-ipfs with a invalid cid, verify that it throws an 'invalid multihash' error
```

Comment:
nit -- an invalid cid :)

Reply:
so the exception actually returrns `invalid multihash`. our naming between cid vs multihash is not consistent at all

Reply:
I meant the `a invalid cid` to `an invalid cid` 🤣

Reply:
fixed!